bftengine/src/bftengine/messages/ReplicaStatusMsg.cpp

```diff
@@ -44,9 +44,9 @@ MsgSize ReplicaStatusMsg::calcSizeOfReplicaStatusMsg(bool viewIsActive,
   else if (viewIsActive)
     return sizeof(ReplicaStatusMsg::Header) + maxNumberOfReplicasBitMaskSize;
   else if (listOfMissingViewChangeMsgForViewChange)
-    return sizeof(ReplicaStatusMsg::Header) + maxNumberOfReplicasBitMaskSize;
+    return sizeof(ReplicaStatusMsg::Header) + (2 * maxNumberOfReplicasBitMaskSize);
```

Comment:
why `2*`?

Reply:
We might receive and exchange Complaints at all times - both when the View is active and if it is not active in both stages (when we are waiting for quorum of View Change messages and for the list of PrePrepares). The highlighted part is the case when our View is inactive and we are waiting for a quorum of View Change messages to proceed. While we do this it is possible to also receive Complaints (from someone who is ahead of us). We need a bit mask to indicate both if we have missing View Change messages from each Replica and what Complaints we currently have for the View again from each peer Replica, therefore 1 maxNumberOfReplicasBitMaskSize for the View Change messages and 1 maxNumberOfReplicasBitMaskSize for the Complaints.